documentation/public/user-guide/05-syntax-trees/03-navigating-with-cursors/index.md

```diff
@@ -0,0 +1,39 @@
+# 5.3. Using Cursors
+
+This guide will walk you through the basics of using a [CST cursor](../../03-concepts/index.md#cursors) in your project.
+Let's start with this source file, that contains three contracts:
+
+```ts title="common.mts"
+--8<-- "documentation/public/user-guide/05-syntax-trees/03-navigating-with-cursors/examples/common.mts"
+```
+
+## Listing Contract Names
+
+The below example uses a cursor to list the names of all contracts in a source file:
+
+```ts title="listing-contract-names.mts"
+--8<-- "documentation/public/user-guide/05-syntax-trees/03-navigating-with-cursors/examples/01-listing-contract-names.test.mts"
+```
+
+## Visiting Only a Sub-tree
+
+Next, we will try to get the names of all contract functions, grouped by the contract name.
+In this case, it is not enough to just visit all instances of `FunctionDefinition` nodes, since we want to exclude the ones that are not part of a contract.
```

Comment:
```suggestion
In this case, it is not enough to just visit all instances of [`FunctionDefinition`](../../../solidity-grammar/02-definitions/08-functions/) nodes, since we want to exclude the ones that are not part of a contract.
```